syntax = "proto3";

package resume;

option go_package = "github.com/iprotoresume/shared/proto";

// ResumeData represents the structured data of a user's resume.
message ResumeData {
  string full_name = 1;
  string email = 2;
  string phone = 3;
  string summary = 4;
  repeated string skills = 5;
  repeated Experience experience = 6;
  repeated Education education = 7;
}

message Experience {
  string title = 1;
  string company = 2;
  string start_date = 3;
  string end_date = 4;
  string description = 5;
}

message Education {
  string degree = 1;
  string institution = 2;
  string graduation_date = 3;
}

// TailorRequest is used to request a tailored resume based on a job description.
message TailorRequest {
  ResumeData original_resume = 1;
  string job_description = 2;
}

message TailorResponse {
  ResumeData tailored_resume = 1;
  string cover_letter = 2;
}

service ResumeService {
  rpc TailorResume (TailorRequest) returns (TailorResponse);
}
